src/mono/mono/metadata/sre-save.c

```diff
@@ -630,17 +630,21 @@ static gboolean
 mono_image_get_method_info (MonoReflectionMethodBuilder *mb, MonoDynamicImage *assembly, MonoError *error)
 {
 	MONO_REQ_GC_UNSAFE_MODE;
+	HANDLE_FUNCTION_ENTER ();
 
```

Comment:
Why is this needed if this function doesn't create any handles ?

Reply:
This method owns the `ReflectionMethodBuilder` and all the pinned handles created from the initialization of the structure. `mono_reflection_methodbuilder_from_method_builder` allocates the handles which live past the method  return.

Reply:
Could you add some comments for that ? This is not the usual way handles are used.